build.sh

```diff
@@ -7,60 +7,21 @@
 ##  the correct version is being used with the correct configuration.       ##
 ##                                                                          ##
 ### ====================================================================== ###
-
-# $Id: build.sh 105735 2010-06-04 19:45:13Z pgier $
-
-# Create the bpa if you can
-BPA=
-uname | grep Linux >> /dev/null
-if [ "$?" -ne "1" ]; then
-	uname -a | grep x86_64 >> /dev/null
-	if [ "$?" -ne "1" ]; then
-		BPA="-Dbpa=centos54x64"
-	else
-		BPA="-Dbpa=centos55x32"
-	fi
-  # This is required for the upgrade of g++ https://issues.jboss.org/browse/JBTM-1787
-  if [ -f /etc/fedora-release ]
-  then
-	uname -a | grep x86_64 >> /dev/null
-	if [ "$?" -ne "1" ]; then
-	    BPA="-Dbpa=fc18x64"
-	fi
-  fi
-fi
+BASH_INTERPRETER=${BASH_INTERPRETER:-${SHELL}}
 
 ORIG_WORKING_DIR=`pwd`
 PROGNAME=`basename $0`
 DIRNAME=`dirname $0`
 GREP="grep"
 ROOT="/"
 
-# Ignore user's MAVEN_HOME if it is set (M2_HOME is unsupported since Apache Maven 3.5.0)
-unset M2_HOME
-unset MAVEN_HOME
-
-JAVA_VERSION=$(java -version 2>&1 | grep "\(java\|openjdk\) version" | cut -d\  -f3 | tr -d '"' | tr -d '[:space:]'| awk -F . '{if ($1==1) print $2; else print $1}')
-
-if [ $JAVA_VERSION -eq "9" ]; then
-  MAVEN_OPTS="$MAVEN_OPTS --add-modules java.corba"
-  MAVEN_OPTS="$MAVEN_OPTS --add-modules java.xml.bind"
-  MAVEN_OPTS="$MAVEN_OPTS --add-modules java.xml.ws"
-  export MAVEN_OPTS
-fi
-
-if [ -z "$MAVEN_OPTS" ]
-then
-	if [ $JAVA_VERSION -ge "9" ]; then
-		MAVEN_OPTS="$MAVEN_OPTS -Xms1303m -Xmx1303m"
```

Comment:
I agree with the bulk of the changes in this file as BPA is not needed and nor is JDK 9 support itself.

From what I can see here, unless it is set somewhere else, I think with this change we are no longer setting our "preferred" Xms Xmx settings? Do I follow correctly? I have a remark on PRESERVE_WORKING_DIR I will mention later.